tt_eager/tensor/tensor.hpp

```diff
@@ -29,8 +29,7 @@ class Tensor {
         // ======================================================================================
         //                                  Hi Level APIs
         // ======================================================================================
-        Tensor(const Storage& storage, const Shape& shape, DataType dtype, Layout layout, std::optional<ShardSpec> shard_spec);
-        Tensor(const Storage& storage, const Shape& shape, DataType dtype, Layout layout);
+        Tensor(const Storage& storage, const Shape& shape, DataType dtype, Layout layout, std::optional<ShardSpec> shard_spec = std::nullopt);
```

Comment:
Did you consider creating a custom `ShardedTensor` type so that sharded / non-sharded code can be separated?
If there was a different `sharded` Tensor type, you wouldn't need to make all these `_sharded` functions (function signatures can be same as regular `Tensor` functions)